### ISSUE TITLE
Error in cca, all row sums must be >0 ???

### ISSUE THREAD

**Issue by NIOO-QingZ** [vegandevs/vegan#337]
Hi,

I met an error when applying cca to my data:

```r
Error in cca.default(d$X, d$Y, d$Z) : 
  all row sums must be >0 in the community data matrix
```

I don't understand why all row sums must be > 0. My data set has a lot of rows whose sum is negative after I scaled it. 
Thanks in advance for any help!

Best
Qing

**Comment by jarioksa** [comment]
I think you should read some textbook about correspondence analysis. It is based on weighted Chi-square distance, and Chi-square is only defined for non-negative data. We are a bit more relaxed, and allow also negative data as long as the column and row sums are positive. However, it is mathematically impossible to have Chi-square distance for negative data or ordination with negative weights (row and column sums are used as weights).

**Comment by NIOO-QingZ** [comment]
Thanks for your response!!!
I went through the correspondence analysis , you are indeed right! cca is based on Chi-square test, but in Chi-square test data has to be positive.
I can use rda which is based on F-test which could work with negative values, because its mathematic function could handle with negative value.

My negative values are derived from scale in which I centered the data. If I set center=False, the negative values could also be avoided.
